lib/matobj.gi

```diff
@@ -1568,3 +1568,394 @@ InstallMethod( \[\,\]\:\=, "for a matrix object, two positions, and an object",
+##  <A>elm</A> from the left in-place.
+##  <P/>
+##  <C>MultMatrixRowLeft</C> is a synonym for <C>MultMatrixRow</C>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixRow, "for a mutable matrix object, one row number, and a scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        ErrorNoReturn("the second argument <row> has to fulfill 0 < row <= NrRows(mat) ");
+    fi;
+  
+    for i in [1..NrCols(mat)] do
+        mat[row,i] := scalar * mat[row,i];
+    od;
+    
+  end );
+
+
+############################################################################
+##
+#O  MultMatrixColumn( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixColumn">
+##  <ManSection>
+##  <Oper Name="MultMatrixColumn" Arg='mat,i,elm' Label=""/>
+##  <Oper Name="MultMatrixColumnRight" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  Multiplies the <A>i</A>-th column of the mutable matrix <A>mat</A> with the scalar 
+##  <A>elm</A> from the right in-place.
+##  <P/>
+##  <C>MultMatrixColumnRight</C> is a synonym for <C>MultMatrixColumn</C>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixColumn, "for a mutable matrix object, one column number, and a scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, column, scalar )
+    local i;
+    
+    # Checks
+    if not( 0 < column and column <= NrCols(mat) ) then
+        ErrorNoReturn("the second argument column has to fulfill 0 < column <= NrCols(mat) ");
+    fi;
+  
+    for i in [1..NrRows(mat)] do
+        mat[i,column] := mat[i,column] * scalar;
+    od;
+    
+  end );
+  
+
+############################################################################
+##
+#O  MultMatrixRowRight( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixRowRight">
+##  <ManSection>
+##  <Oper Name="MultMatrixRowRight" Arg='mat,i,elm' Label=""/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  Multiplies the <A>i</A>-th row of the mutable matrix <A>mat</A> with the scalar 
+##  <A>elm</A> from the right in-place.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+InstallMethod( MultMatrixRowRight, "for a mutable matrix object, one row number, and a scalar",
+  [ IsMatrixObj and IsMutable, IsInt, IsObject ],
+  function( mat, row, scalar )
+      local i;
+    
+    # Checks
+    if not( 0 < row and row <= NrRows(mat) ) then
+        ErrorNoReturn("the second argument row has to fulfill 0 < row <= NrRows(mat) ");
+    fi;
+  
+    for i in [1..NrCols(mat)] do
+        mat[row,i] := mat[row,i] * scalar;
+    od;
+    
+  end );
+
+############################################################################
+##
+#O  MultMatrixColumnLeft( <M>, <row>, <scalar> )
+##
+##  <#GAPDoc Label="MultMatrixColumnLeft">
```

Comment:
What exactly is the policy on that? We oriented ourselves on the code that already existed. There are quite a few documentation comments in `matobj.gi`. The same holds for the `#O` and `#M` lines. They are everywhere so we assumed they are for something we don't know about. Can we remove them then evrywhere in the matobj stuff? Or even in the whole library? Maybe in a new PR?

Reply:
In our code I already removed all lines starting with #O or #M. Not in the entire `matobj.gi` file, though.

Reply:
The documentation comments for newly declared operations, attributes, etc. all should be next to the declaration of that operation, attribute, etc., not next to one (arbitrary) method. 

In contrast to this, the GAPDoc comments inside `matobj.gi` are all for specific methods, or they document concepts. E.g. the first one starting in line 21, with label "MatrixObjCompare", explains how MatrixObj instances are supposed to be compared, and to that end also "documents" the `\=` method "for two vector objects". Since this is about a specific *method* (or rather, a bunch of them), and not a general operation, it makes place to put it right next to where those methods are defined.